src/AppInstallerCLICore/Workflows/ConfigurationFlow.cpp

```diff
@@ -837,17 +846,264 @@ namespace AppInstaller::CLI::Workflow
+                }
+                else
+                {
+                    description = L"Configure " + resourceNameWide;
+                }
+
+                directives.Insert(s_Directive_Description, PropertyValue::CreateString(description));
+                unit.Metadata(directives);
+
+                // Call processor to get settings for the unit.
+                auto getResult = GetUnitSettings(context, unit);
+                winrt::hresult resultCode = getResult.ResultInformation().ResultCode();
+                if (FAILED(resultCode))
+                {
+                    // Retry if it fails with not found in the case the module is a pre-released one.
+                    bool isPreRelease = false;
+                    if (resultCode == WINGET_CONFIG_ERROR_UNIT_NOT_FOUND_REPOSITORY)
+                    {
+                        directives.Insert(s_Directive_AllowPrerelease, PropertyValue::CreateBoolean(true));
+                        unit.Metadata(directives);
+
+                        auto preReleaseResult = GetUnitSettings(context, unit);
+                        if (SUCCEEDED(preReleaseResult.ResultInformation().ResultCode()))
+                        {
+                            isPreRelease = true;
+                            getResult = preReleaseResult;
+                        }
+                        else
+                        {
+                            AICLI_LOG(Config, Error, << "Failed Get allowing prerelease modules");
+                            LogFailedGetConfigurationUnitDetails(unit, preReleaseResult.ResultInformation());
+                        }
+                    }
+
+                    if (!isPreRelease)
+                    {
+                        OutputUnitRunFailure(context, unit, getResult.ResultInformation());
+                        THROW_HR(WINGET_CONFIG_ERROR_GET_FAILED);
+                    }
+                }
+
+                unit.Settings(getResult.Settings());
+
+                // GetUnitSettings will set it to Inform.
+                unit.Intent(ConfigurationUnitIntent::Apply);
+
+                // Add dependency if needed.
+                if (dependantUnit.has_value())
+                {
+                    auto dependencies = winrt::single_threaded_vector<winrt::hstring>();
+                    dependencies.Append(dependantUnit.value().Identifier());
+                    unit.Dependencies(std::move(dependencies));
+                }
+
+                return unit;
+            }
+
+            return {};
+        }
     }
 
     void CreateConfigurationProcessor(Context& context)
     {
         auto progressScope = context.Reporter.BeginAsyncProgress(true);
         progressScope->Callback().SetProgressMessage(Resource::String::ConfigurationInitializing());
 
-        ConfigurationProcessor processor{ CreateConfigurationSetProcessorFactory(context)};
+        ConfigurationProcessor processor{ anon::CreateConfigurationSetProcessorFactory(context)};

```

Comment:
Is there a reason for explicitly naming the anonymous namespace? I haven't seen this done before

Reply:
I've learned that anonymous namespace does not output type information into the PDB in a way that debuggers (at least VS and WinDBG) can find currently, making it much harder to debug. With a named namespace, the derived type information is found and so the debugger can show the member fields of a derived type (vtable -> type mapping). One can manually do all of this with enough effort, and maybe eventually the debuggers will support it directly, but at this time I see simply naming the namespace as the easiest answer.